### PR TITLE
Biosamples Converter

### PR DESCRIPTION
**Pull request by MightyAx**
Move BioSamples conversion functionality our of BioSample Service and into a new BioSamples converter class to match out other implementations of conversion.

- [x] Save units into column map
- [x] Use units in BioSamples conversion
- [x] Add Unit Tests

### CODE REVIEW

**Comment by MightyAx** [comment]
[Seperate BioSamples converter from BioSamples Service](https://trello.com/c/zt5aSjXG/32-seperate-biosamples-converter-from-biosamples-service)

File: test/unit/conversion/biosamples/test_biosamples_converter.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        ]
+        entity = Entity('sample', 'sample1', attributes)
+        sample = self.converter.convert_sample(entity)
+        
+        self.assertEqual(attributes['sample_title'], sample.name)
+        self.assertEqual(attributes['tax_id'], sample.ncbi_taxon_id)
+        self.assertEqual(attributes['scientific_name'], sample.species)
+        self.assertAttributesEqual(expected_attributes, sample.attributes)
+    
+    def assertAttributesEqual(self, expected: List[Attribute], actual: List[Attribute]):
```
Comment (ke4) [inline test/unit/conversion/biosamples/test_biosamples_converter.py:60]: I suggest to make this utility method private

File: test/unit/conversion/biosamples/test_biosamples_converter.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self.assertEqual(len(expected), len(actual))
+        for index in range(0, len(expected)):
+            self.assertAttributeEqual(expected[index], actual[index])
+    
+    def assertAttributeEqual(self, expected: Attribute, actual: Attribute):
```
Comment (ke4) [inline test/unit/conversion/biosamples/test_biosamples_converter.py:65]: I suggest to make this utility method private, too

**Review by ke4 — APPROVED** [review]
Just 2 very tiny comment otherwise 💯  👍

**Comment by mshadbolt** [comment]
I tried to submit a spreadsheet with a few attributes that had units but submission doesn't seem to be working properly. It says it wants the biosamples accession. let me know if you want me to send you any of the other files.

The errors I get in the issues.json are:

```
{
  "6": [
    "Error: sample.sample_accession BioSamples Error: Expecting value: line 1 column 1 (char 0)"
  ],
  "7": [
    "Error: sample.sample_accession BioSamples Error: Expecting value: line 1 column 1 (char 0)"
  ]
}
```

**Comment by mshadbolt** [comment]
@MightyAx and I did some debugging and we figured out it was something weird going on with the aap domain. I tried again and it looks like it worked perfectly. Thanks so much for turning this around so quickly!

**Review by mshadbolt — APPROVED** [review]
Works for me!

**Pull request merged**